pkg/trace/obfuscate/sql.go

```diff
@@ -151,22 +152,58 @@ func (f *groupingFilter) Reset() {
 // Process the given SQL or No-SQL string so that the resulting one is properly altered. This
 // function is generic and the behavior changes according to chosen tokenFilter implementations.
 // The process calls all filters inside the []tokenFilter.
-func obfuscateSQLString(in string) (string, error) {
-	tokenizer := NewSQLTokenizer(in)
+func (o *Obfuscator) obfuscateSQLString(in string) (string, error) {
+	ignoreEscapes := o.ShouldIgnoreEscapes()
+	tokenizer := NewSQLTokenizer(in, ignoreEscapes)
 	filters := []tokenFilter{&discardFilter{}, &replaceFilter{}, &groupingFilter{}}
+
+	out, err := attemptObfuscation(tokenizer, filters)
+	if err != nil {
+		// If the tokenizer failed, but saw an escape character in the process
+		// retry with opposite setting.
+		if tokenizer.HasSeenEscape() {
+			tokenizer = NewSQLTokenizer(in, !ignoreEscapes)
+			var err2 error
+			out, err2 = attemptObfuscation(tokenizer, filters)
+			if err2 == nil {
+				// If the second attempt succeeded, make the config change
+				err = nil
+				o.SetIgnoreEscapes(!ignoreEscapes)
+			}
+		}
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	if len(out) == 0 {
+		return "", errors.New("result is empty")
+	}
+
+	return out, nil
```

Comment:
If you move the "result is empty" check into `attemptObfuscation` and apply my other comments we can replace all of this with:
```go
return out, err
```
Unless I've missed something...